src/step-function.ts

```diff
@@ -466,6 +468,110 @@ export namespace $SFN {
       });
     },
   });
+
+  type AWSServiceClass = { new (): AWSService };
+
+  /**
+   * First we have to extract the names of all Services in the v2 AWS namespace
+   *
+   * @returns "AccessAnalyzer" | "Account" | ... | "XRay"
+   */
+  type ServiceKeys = {
+    [K in keyof typeof AWS]: typeof AWS[K] extends AWSServiceClass ? K : never;
+  }[keyof typeof AWS];
+
+  /**
+   * We define this to omit any non-funciton types from service
+   * instances. e.g. `apiVersions` and `config`
+   */
+  type ServiceInstance<T extends AWSServiceClass> = {
+    [K in FunctionKeys<InstanceType<T>>]: InstanceType<T>[K];
+  };
+
+  export const SDK: {
+    [serviceName in ServiceKeys]: ServiceInstance<typeof AWS[serviceName]>;
+  } = new Proxy({} as any, {
+    get(_, serviceName: ServiceKeys) {
+      const client = new AWS[serviceName]();
+      const sdkIntegrationServiceName = mapSDKServiceName(serviceName);
+
+      return new Proxy(client as any, {
+        get: (_, prop: string) => {
+          return makeIntegration<"$AWS.SDK", (input: any) => Promise<any>>({
+            kind: "$AWS.SDK",
+            asl: (call, context) => {
+              const input = call.args[0]?.expr;
+              if (!isObjectLiteralExpr(input)) {
+                throw new SynthError(
+                  ErrorCodes.Expected_an_object_literal,
+                  `First argument ('input') into $AWS.SDK must be an object.`
+                );
+              }
```

Comment:
This limitation is unnecessary. @thantos can confirm, but I believe we only needed that constraint before because we were extracting the Table from the ObjectLiteral. Step Functions can evaluate arbitrary inputs so it doesn't matter in the case when passing a string.

Reply:
Left a comment on the newest version, should be able to support any input object.